### PR TITLE
Test : Added tests for insights-client --unregister option

### PR DESCRIPTION
**Pull request by Archana-PandeyM**

### CODE REVIEW

File: integration-tests/test_unregister.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+def test_unregister(insights_client):
+    """Test insights-client --unregister after registering it
+
+    If the unregistration is successful, `unregister` method returns True
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    registration_status = insights_client.run("--unregister")
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:14]:
```suggestion
    registration_status = insights_client.unregister()
```

File: integration-tests/test_unregister.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    assert (
+        "Successfully unregistered from the Red Hat Insights Service"
+        in registration_status.stdout
+    )
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:18]: after this, you can also assert that is `not is_registered`

File: integration-tests/test_unregister.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    """Test insights-client --unregister on unregistered system
+
+    If the unregistration is successful, `--unregister` returns True on first attempt and
+    false on subsequent attempts.
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    # unregister once
+    registration_status = insights_client.run("--unregister")
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:31]: ditto (`unregister()`)

File: integration-tests/test_unregister.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    assert not insights_client.is_registered
+    assert (
+        "Successfully unregistered from the Red Hat Insights Service"
+        in registration_status.stdout
+    )
+    # unregister twice
+    registration_status = insights_client.run("--unregister", check=False)
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:38]: ditto (`unregister()`)

**Review by ptoscano — CHANGES_REQUESTED** [review]
Generally LGTM; added a couple of notes.

----

Do we really need two different tests? After all, `test_unregister_twice` is exactly `test_unregister` plus an extra check; IMHO it would seem easier to do only one test (`test_unregister`) that check for unregistration, also if done twice.

File: integration-tests/test_unregister.py (hunk 4 above)
Comment (ptoscano) [reply integration-tests/test_unregister.py:38]: ah scratch this, you cannot pass any additional parameters to `unregister()`

File: integration-tests/test_unregister.py (hunk 1 above)
Comment (Archana-PandeyM) [reply integration-tests/test_unregister.py:14]: In my opinion I do not see a problem with this, also it looks symmetrical throughout test(aesthetically )

File: integration-tests/test_unregister.py (hunk 2 above)
Comment (Archana-PandeyM) [reply integration-tests/test_unregister.py:18]: ok I wil add this.

**Comment by Archana-PandeyM** [comment]
> Generally LGTM; added a couple of notes.
> 
> Do we really need two different tests? After all, `test_unregister_twice` is exactly `test_unregister` plus an extra check; IMHO it would seem easier to do only one test (`test_unregister`) that check for unregistration, also if done twice.

I think these two are different tests and hence two tests.

**Comment by jstavel** [comment]
looks good to me. /lgtm

**Comment by ptoscano** [comment]
> I think these two are different tests and hence two tests.

While that is true for a "scenario" POV, in practice running `test_unregister_twice` alone is enough to check that everything related to the unregistration works, so `test_unregister` is redundant.

**Comment by Archana-PandeyM** [comment]
> > I think these two are different tests and hence two tests.
> 
> While that is true for a "scenario" POV, in practice running `test_unregister_twice` alone is enough to check that everything related to the unregistration works, so `test_unregister` is redundant.

In my opinion two different tests are needed because the intent of both tests are different.

**Comment by zhangqianqian** [comment]
I think the 2 cases are different.
test_unregister is for the basic function
test_unregister_twice is negative testing: check what happens when run "insights-client --unregister "on a unregistered system. So I agreee with Archana.

File: integration-tests/test_unregister.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+def test_unregister(insights_client):
+    """Test insights-client --unregister after registering it
+
+    If the unregistration is successful, `unregister` method returns True
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    registration_status = insights_client.run("--unregister")
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:14]: this variable is not used -- simply do not assign the result of `run()`

File: integration-tests/test_unregister.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    """Test insights-client --unregister on unregistered system
+
+    If the unregistration is successful, `--unregister` returns True on first attempt and
+    false on subsequent attempts.
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    # unregister once
+    registration_status = insights_client.run("--unregister")
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:32]: ditto (unused variable)

File: integration-tests/test_unregister.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    assert not insights_client.is_registered
+    assert (
+        "Successfully unregistered from the Red Hat Insights Service"
+        in registration_status.stdout
+    )
+    # unregister twice
+    registration_status = insights_client.run("--unregister", check=False)
```
Comment (ptoscano) [inline integration-tests/test_unregister.py:39]: maybe `s/registration_status/unregistration_status/` for clarity?

**Review by ptoscano — CHANGES_REQUESTED** [review]
OK, I'm still not convinced about the duplication of tests, however I'm fine in keeping it for now. Please then amend the commit message to explain why both are needed, even though one test does the job also of another test: this way, it will be less confusing when browsing through the git history in the future.

I added a couple of minor notes that I missed in the previous review.

After these changes, please rebase against the `master` branch.

Thanks!

File: integration-tests/test_unregister.py (hunk 5 above)
Comment (Archana-PandeyM) [reply integration-tests/test_unregister.py:14]: I did not understand ? It is used in assert statement.

Comment (ptoscano) [reply integration-tests/test_unregister.py:14]: <facepalm>

ignore this, sorry

File: integration-tests/test_unregister.py (hunk 6 above)
Comment (ptoscano) [reply integration-tests/test_unregister.py:32]: *facepalm*

ignore this, sorry

File: integration-tests/test_unregister.py (hunk 7 above)
Comment (Archana-PandeyM) [reply integration-tests/test_unregister.py:39]: registration status is status of system, I mean status of system. this can be true/false.

Comment (ptoscano) [reply integration-tests/test_unregister.py:39]: well, in this case it is the status of the unregistration; what you describe would be the output of `--status`

**Review by ptoscano — APPROVED** [review]
LGTM, thanks!

waiting for the successful runs of the tests before merging

**Pull request merged**